src/core/wallet/hw-wallet/ledger/transport-factory.ts

```diff
@@ -0,0 +1,34 @@
+import { HWModel, HWConnection } from '../types';
+import { BLE } from './transport/transport-ble';
+import { USB } from './transport/transport-usb';
+import { Platform } from 'react-native';
+
+export class TransportFactory {
+    public static async get(
+        deviceModel: HWModel,
+        connectionType: HWConnection,
+        deviceId?: string
+    ): Promise<Transport> {
+        switch (deviceModel) {
```

Comment:
since we have a config file, where we have all the combinations try to use it :)
You can do something like this:
1. check if deviceModel - connectionType combination is valid, using the config
2. return transport based on connectionType

This way if we only have to change config and this will work